examples/pl_servable_module/production.py

```diff
@@ -0,0 +1,137 @@
+import base64
+import os
+import sys
+from dataclasses import dataclass
+from io import BytesIO
+from os import path
+from typing import Optional
+
+import numpy as np
+import torch
+import torchvision
+import torchvision.models as models
+import torchvision.transforms as T
+from PIL import Image as PILImage
+
+from pytorch_lightning import cli_lightning_logo, LightningDataModule, LightningModule
+from pytorch_lightning.serve import ServableModule, ServableModuleValidator
+from pytorch_lightning.utilities.cli import LightningCLI
+
+DEFAULT_CMD_LINE = (
+    "fit",
+    "--trainer.max_epochs=1",
+    "--trainer.limit_train_batches=15",
+    "--trainer.limit_val_batches=15",
+)
+DATASETS_PATH = path.join(path.dirname(__file__), "..", "..", "Datasets")
+
+
+class LitModule(LightningModule):
+    def __init__(self, name: str = "resnet18"):
+        super().__init__()
+        self.model = getattr(models, name)(pretrained=True)
+        self.criterion = torch.nn.CrossEntropyLoss()
+
+    def training_step(self, batch, batch_idx):
+        inputs, labels = batch
+        outputs = self.model(inputs)
+        loss = self.criterion(outputs, labels)
+        self.log("train_loss", loss)
+        return loss
+
+    def validation_step(self, batch, batch_idx):
+        inputs, labels = batch
+        outputs = self.model(inputs)
+        loss = self.criterion(outputs, labels)
+        self.log("val_loss", loss)
+
+    def configure_optimizers(self):
+        return torch.optim.SGD(self.parameters(), lr=0.001, momentum=0.9)
+
+
+class CIFAR10DataModule(LightningDataModule):
+
+    transform = T.Compose([T.Resize(256), T.CenterCrop(224), T.ToTensor()])
+
+    def train_dataloader(self, *args, **kwargs):
+        trainset = torchvision.datasets.CIFAR10(root=DATASETS_PATH, train=True, download=True, transform=self.transform)
+        return torch.utils.data.DataLoader(trainset, batch_size=2, shuffle=True, num_workers=0)
+
+    def val_dataloader(self, *args, **kwargs):
+        valset = torchvision.datasets.CIFAR10(root=DATASETS_PATH, train=False, download=True, transform=self.transform)
+        return torch.utils.data.DataLoader(valset, batch_size=2, shuffle=True, num_workers=0)
+
+
+@dataclass(unsafe_hash=True)
+class Image:
+
+    height: Optional[int] = None
+    width: Optional[int] = None
+    extension: str = "JPEG"
+    mode: str = "RGB"
+    channel_first: bool = False
+
+    def deserialize(self, data: str) -> torch.Tensor:
+        encoded_with_padding = (data + "===").encode("ascii")
```

Comment:
what is this padding used for?

Reply:
This was used in Flash. I kept it as it was.